src/curvature/functions.jl

```diff
@@ -16,9 +16,9 @@ function jacobians(curvature::CurvatureInterface, X::AbstractArray)
     # Output:
     ŷ = nn(X)
     # Jacobian:
-    𝐉 = jacobian(() -> nn(X), Flux.params(nn))
-    𝐉 = permutedims(reduce(hcat, [𝐉[θ] for θ in curvature.params]))
-    return 𝐉, ŷ
+    𝐉 = jacobian(() -> nn(X), Flux.params(nn))                                # differentiates f with regards to the model parameters
```

Comment:
How does this work? We pass a function of zero args that returns the NN output for the input? But then if it was always a const value being returned, then the Jacobian would be zero. So how does Zygote compute this? By permuting parameters by an epsilon-value, and then seeing how `nn(X)` changes?

```
jacobian(f, args...) -> Tuple

For each array a ∈ args this returns a matrix with Ja[k,i] = ∂y[k]/∂a[i] where y = f(args...) is usually a vector. Arrays of higher dimension are treated like vec(a), or vec(y) for output.
```

Reply:
I think this has less to do with us documenting it and more with understanding how Zygote works. Zygote is still able to compute the Jacobian by analyzing the computation graph of the neural network and its parameters even though the function passed to it has no arguments.

Reply:
I added a bit of bonus explanation, but most likely we will update it when we fully understand how Zygote works.